discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -18,65 +18,136 @@
+    if latest_slot is None:
+        latest_slot = 0
+
+    logger.info(f"index_solana_plays.py | returning {latest_slot} for highest slot")
+    return latest_slot
+
 def process_solana_plays(solana_client):
-    redis = index_solana_plays.redis
     db = index_solana_plays.db
 
-    slot_from = get_pickled_key(redis, SOL_PLAYS_REDIS_KEY)
-    if not slot_from:
-        slot_from = solana_client.get_slot()["result"]
-        pickle_and_set(redis, SOL_PLAYS_REDIS_KEY, slot_from)
-
-    # slot_from = solana_client.get_slot()["result"]
-    transaction = solana_client.get_confirmed_signature_for_address2(
-        TRACK_LISTEN_PROGRAM, limit=1)
-    if transaction["result"][0]["slot"] > slot_from:
-        slot_from = transaction["result"][0]["slot"]
-        pickle_and_set(redis, SOL_PLAYS_REDIS_KEY, slot_from)
-        # TODO: What happens if there is >1 tx since the last iteration here?
-        tx_info = solana_client.get_confirmed_transaction(
-            transaction["result"][0]["signature"]
-        )
-        if SECP_PROGRAM in tx_info["result"]["transaction"]["message"]["accountKeys"]:
-            audius_program_index = tx_info["result"]["transaction"]["message"]["accountKeys"].index(
-                TRACK_LISTEN_PROGRAM
-            )
-            for instruction in tx_info["result"]["transaction"]["message"]["instructions"]:
-                if instruction["programIdIndex"] == audius_program_index:
-                    hex_data = binascii.hexlify(
-                        bytearray(list(base58.b58decode(instruction["data"])))
-                    )
+    # Highest currently processed slot in the DB
+    latest_processed_slot = get_latest_slot(db)
+    logger.info(f"index_solana_plays.py | latest used slot: {latest_processed_slot}")
 
-                    l1 = int(hex_data[2:4], 16)
-                    start_data1 = 10
-                    end_data1 = l1 * 2 + start_data1
+    # Loop exit condition
+    intersection_found = False
 
-                    l2 = int(hex_data[end_data1:end_data1 + 2], 16)
-                    start_data2 = end_data1 + 8
-                    end_data2 = l2 * 2 + start_data2
+    # List of signatures that will be populated as we traverse recent operations
+    transaction_signatures = []
 
-                    l3 = int(hex_data[end_data2:end_data2 + 2], 16)
-                    start_data3 = end_data2 + 8
-                    end_data3 = l3 * 2 + start_data3
+    last_tx_signature = None
 
-                    user_id = codecs.decode(hex_data[start_data1:end_data1], "hex")
-                    track_id = codecs.decode(hex_data[start_data2:end_data2], "hex")
-                    source = codecs.decode(hex_data[start_data3:end_data3], "hex")
+    # Traverse recent records until an intersection is found with existing Plays table
+    while not intersection_found:
+        # TODO: Is there any optimization around this limit value?
+        transactions_history = solana_client.get_confirmed_signature_for_address2(
```

Comment:
Is there any optimization around this limit value? Max is 1k

Reply:
I think this value should be slightly higher, perhaps let's test with 100 and 1000 to get a sense of things?